torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -530,10 +559,20 @@ class TensorExprFuser {
       REQ(producer->input(0)->node()->kind() == prim::ListConstruct);
       REQ(producer->input(0)->uses().size() == 1);
       REQ(producer->input(1)->node()->kind() == prim::Constant);
+      auto const& listConstruct = producer->input(0)->node();
+      for (auto const& input : listConstruct->inputs()) {
+        REQ(isFusableOnDevice(input->node()));
```

Comment:
Not necessarily to do in this PR, but we also need to be checking that we're pulling in values/nodes with the same device.

Reply:
I'll have to think a bit more about restricting merges to same device.  I think that problem exists even without this PR, so I'm going to go ahead and land this one while I work on same-device.